unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenPercentageController.cs

```diff
@@ -0,0 +1,79 @@
+using DCL.Controllers;
+using System;
+using UnityEngine;
+
+namespace DCL.LoadingScreen
+{
+    /// <summary>
+    /// Loading screen percentage updater. The responsibility of this class is to update the loading bar only
+    /// listening to the destination scene
+    /// </summary>
+    public class LoadingScreenPercentageController : IDisposable
+    {
+
+        private readonly LoadingScreenPercentageView loadingScreenPercentageView;
+        private readonly ISceneController sceneController;
+
+        private Vector2Int currentDestination;
+        private ParcelScene currentScene;
+
+
+        public LoadingScreenPercentageController(ISceneController sceneController, LoadingScreenPercentageView loadingScreenPercentageView)
+        {
+            this.loadingScreenPercentageView = loadingScreenPercentageView;
+            this.sceneController = sceneController;
+
+            loadingScreenPercentageView.gameObject.SetActive(false);
+
+            sceneController.OnNewSceneAdded += SceneController_OnNewSceneAdded;
+        }
+
+        public void Dispose()
+        {
+            sceneController.OnNewSceneAdded -= SceneController_OnNewSceneAdded;
+            if (currentScene)
+            {
+                currentScene.sceneLifecycleHandler.sceneResourcesLoadTracker.OnResourcesStatusUpdate -= ResourcesStatusUpdate;
+                currentScene.sceneLifecycleHandler.OnStateRefreshed -= Scene_OnStateRefreshed;
+            }
+        }
+
+        private void SceneController_OnNewSceneAdded(IParcelScene scene)
+        {
+            currentScene = scene as ParcelScene;
```

Comment:
Casting violates "Liskov Substitution" principle: we should operate with interfaces instead and do not rely on the underlying implementation

Reply:
Found a better event that I could suscribe without casting :)